server/sonar-db-dao/src/main/java/org/sonar/db/metric/MetricValidator.java

```diff
@@ -60,4 +61,16 @@ static String validateDescription(@Nullable String description) {
 
     return description;
   }
+
+  @CheckForNull
+  public static String checkMetricDomain(@Nullable String domain) {
+    if (domain == null) {
+      return null;
+    }
+
+    checkArgument(domain.length() <= MAX_DOMAIN_LENGTH, "Metric domain length (%s) is longer than the maximum authorized (%s). '%s' was provided.",
+      domain.length(), MAX_DOMAIN_LENGTH, domain);
```

Comment:
This does not yet seem to work:
```
$ curl -s -u admin:admin -XPOST "localhost:9000/api/metrics/update?id=144&domain=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
{"errors":[{"msg":"An error has occurred. Please contact your administrator"}]}
```

Reply:
Using exactly your curl command, I have the expected message…

Reply:
I rebuilt the latest state of the branch, and now it also works fine for me. Sorry for the noise.